src/plugins/orangehrmRecruitmentPlugin/Controller/ApplicantController.php

```diff
@@ -0,0 +1,317 @@
+        CandidateHistory $applicantHistory,
+        int $applicantId,
+        int $action
+    ): void {
+        $applicantHistory->getDecorator()->setCandidateById($applicantId);
+        $applicantHistory->setAction($action);
+        $applicantHistory->setPerformedDate($this->getDateTimeHelper()->getNow());
+    }
+
+
+    /**
+     * @return ValidationDecorator
+     */
+    protected function getValidationDecorator(): ValidationDecorator
+    {
+        if (!$this->validationDecorator instanceof ValidationDecorator) {
+            $this->validationDecorator = new ValidationDecorator();
+        }
+        return $this->validationDecorator;
+    }
+
+    /**
+     * @return ParamRuleCollection|null
+     */
+    protected function getParamRuleCollection(): ?ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                self::PARAMETER_RESUME,
+                new Rule(Rules::BASE_64_ATTACHMENT)
+            ),
+            new ParamRule(
+                self::PARAMETER_FIRST_NAME,
+                new Rule(Rules::STRING_TYPE)
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_MIDDLE_NAME,
+                    new Rule(Rules::STRING_TYPE)
+                ),
+                true
+            ),
+            new ParamRule(
+                self::PARAMETER_LAST_NAME,
+                new Rule(Rules::STRING_TYPE)
+            ),
+            new ParamRule(
+                self::PARAMETER_EMAIL,
+                new Rule(Rules::EMAIL)
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_CONTACT_NUMBER,
+                    new Rule(Rules::PHONE)
+                )
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_VACANCY_ID,
+                    new Rule(Rules::POSITIVE)
+                )
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_CONSENT_TO_KEEP_DATA,
+                    new Rule(Rules::STRING_TYPE)
+                )
+            ),
+        );
+    }
+
+    /**
+     * @param CandidateAttachment $applicantAttachment
+     * @param int $applicantId
+     * @param array $resume
+     */
+    private function setCandidateAttachment(CandidateAttachment $applicantAttachment, int $applicantId, array $resume)
+    {
+        $applicantAttachment->getDecorator()->setCandidateById($applicantId);
+        $this->setBase64Attachment($applicantAttachment, $resume);
+    }
+
+    /**
+     * @param CandidateAttachment $applicantAttachment
+     * @param array $resume
+     */
+    private function setBase64Attachment(CandidateAttachment $applicantAttachment, array $resume): void
```

Comment:
```suggestion
    private function setBase64Attachment(CandidateAttachment $applicantAttachment, Base64Attachment $resume): void
```
:thinking:  and call getter method of `Base64Attachment`

Reply:
Done